pkg/proxy/apis/config/validation/validation_test.go

```diff
@@ -816,3 +817,176 @@ func TestValidateKubeProxyExcludeCIDRs(t *testing.T) {
 		}
 	}
 }
+
+func TestValidationYAML(t *testing.T) {
```

Comment:
@liggitt @luxas Though "Validation testing using YAML files in `testdata/`" is listed in https://github.com/kubernetes/enhancements/blame/master/keps/sig-cluster-lifecycle/wgs/0032-create-a-k8s-io-component-repo.md#L187, I'm not sure if it adds that much value compared to just adding more/finer grained test cases to the object validation unit tests we already write (without going through the decode, like https://github.com/kubernetes/kubernetes/pkg/proxy/apis/config/validation/validation_test.go). 

WDYT? Was there a rationale for having these as well?

Reply:
Tests for `yaml -> decode`, `yaml -> defaulting`, and `yaml -> conversion` have value.

For validation, passing in structs to exercise validation scenarios is usually cleaner.

Reply:
Sounds good. Shall we check whether the struct-based validation tests cover the same cases, add cases to struct-based validation tests where they are missing, and remove the yaml-based validation tests for now?

Reply:
unless the yaml-based tests are checking something specific to decoding (like strict vs lenient decoding, duplicate/unknown key handling, etc), switching them to struct-based and filling any gaps there would make sense to me